modules/storage-ephemeral-storage-manage.adoc

```diff
@@ -1,8 +1,57 @@
+
+You can manage local ephemeral storage by specifying requests and limits. Each container in a pod can specify the following:
+
+* `spec.containers[].resources.limits.ephemeral-storage`
+* `spec.containers[].resources.requests.ephemeral-storage`
+
+Limits and requests for ephemeral storage are measured in byte quantities. You can express storage as a plain integer or as a fixed-point number using one of these suffixes: E, P, T, G, M, k. You can also use the power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki. For example, the following quantities all represent approximately the same value: 128974848, 129e6, 129M, and 123Mi. The case of the suffixes is significant. If you specify 400m of ephemeral storage, this requests 0.4 bytes, rather than 400 mebibytes (400Mi) or 400 megabytes (400M), which was probably what was intended.
+
+The following example shows a pod with two containers. Each container requests 2GiB of local ephemeral storage. Each container has a limit of 4GiB of local ephemeral storage. Therefore, the pod has a request of 4GiB of local ephemeral storage, and a limit of 8GiB of local ephemeral storage.
+
+[source, yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: frontend
+spec:
+  containers:
+  - name: app
+    image: images.my-company.example/app:v4
+    resources:
+      requests:
+        ephemeral-storage: "2Gi" <1>
+      limits:
+        ephemeral-storage: "4Gi" <2>
+    volumeMounts:
+    - name: ephemeral
+      mountPath: "/tmp"
+  - name: log-aggregator
+    image: images.my-company.example/log-aggregator:v6
+    resources:
+      requests:
+        ephemeral-storage: "2Gi" <1>
+      limits:
+        ephemeral-storage: "4Gi" <2>
+    volumeMounts:
+    - name: ephemeral
+      mountPath: "/tmp"
+  volumes:
+    - name: ephemeral
+      emptyDir: {}
+----
+<1> Request for local ephemeral storage.
+<2> Limit for local ephemeral storage.
+
+
+This setting in the pod spec affects how the scheduler makes a decision on scheduling pods, and also how kubelet evict pods. First of all, the scheduler ensures that the sum of the resource requests of the scheduled containers is less than the capacity of the node. In this case, the pod can be assigned to a node only if its available ephemeral storage (allocatable resource) is more than 4GiB.
+
+Secondly, at the container level, since one of the the containers sets resource limit, kubelet eviction manager measures the disk usage of this container and evicts the pod if the storage usage of the first container exceeds its limit (8GiB). At the pod level, kubelet works out an overall pod storage limit by adding up the limits of all the containers in that pod. In this case, the total storage usage at the pod level is the sum of the disk usage from all containers plus the pod's `emptyDir` volumes. If this total usage exceeds the overall pod storage limit (8GiB), then the kubelet also marks the pod for eviction.
```

Comment:
Typo with two "the" -> since one of the the containers

Why do we say **one** of the container sets resource limit? Looking at the snippet both containers have a 4Gi limit.

Reply:
@gcharot I think that I've fixed it.

Reply:
Hi @lpettyjo @gcharot , yeah it seems the `since one of the the containers sets resource limit` is not keep the same with our example. We'd better keep the same with the blog. The limit and request size could use different values. Others we could reuse.

https://kubernetes.io/blog/2022/09/19/local-storage-capacity-isolation-ga/#setting-requests-and-limits-for-local-ephemeral-storage

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: frontend
spec:
  containers:
  - name: app
    image: images.my-company.example/app:v4
    resources:
      requests:
        ephemeral-storage: "8Gi"
      limits:
        ephemeral-storage: "12Gi"
    volumeMounts:
    - name: ephemeral
      mountPath: "/tmp"
  - name: log-aggregator
    image: images.my-company.example/log-aggregator:v6
    resources:
      requests:
        ephemeral-storage: "2Gi"
    volumeMounts:
    - name: ephemeral
      mountPath: "/tmp"
  volumes:
    - name: ephemeral
      emptyDir: 
        sizeLimit: 5Gi
```
In additin, in this template also has the sizeLimit if we use it also need to clarify.
```doc
Last, in this example, emptyDir volume sets its sizeLimit to 5Gi. It means that if this pod's emptyDir used up more local storage than 5GiB, the pod will be evicted from the node.
```

Reply:
@gcharot @Phaow OK, I think I have the example request/limit values matching the text. I didn't include sizeLimit in the example, nor text, because as far as I know it's not in scope for this feature.

Reply:
Thanks for the update! LGTM